cunumeric/deferred.py

```diff
@@ -1740,6 +1766,28 @@ def put(self, indices: Any, values: Any, check_bounds: bool) -> None:
         if self_tmp is not self:
             self.copy(self_tmp, deep=True)
 
+    @auto_convert("mask", "values")
+    def putmask(self, mask: Any, values: Any) -> None:
+        is_scalar_value = False
+        if values.shape != self.shape and values.size == 1:
+            if values.shape == ():
+                values = values._convert_future_to_regionfield(True)
+            is_scalar_value = True
+
+        task = self.context.create_task(CuNumericOpCode.PUTMASK)
+        task.add_input(self.base)
+        task.add_input(mask.base)
+        task.add_input(values.base)
+        task.add_output(self.base)
+        task.add_scalar_arg(is_scalar_value, bool)  # value is scalar
```

Comment:
again, is there a reason that you don't do numpy broadcasting on the scalar value? that would make the code a lot simpler.

Reply:
makes sense. replaced the logic with broadcasting